### ISSUE TITLE
Current docker based build is broken

### ISSUE THREAD

**Issue by EugenMayer** [ianw1974/zimbra-build-scripts#58]
Building within the docker-image i get 

```bash
/home/git/zimbra/zm-mailbox/build.xml:13: The following error occurred while executing this line:
/home/git/zimbra/zm-mailbox/build-common.xml:576: Directory does not exist: /root/.ivy2/cache
```

Should be at least reproduceable when running a Zimbra 10.0.5 build

**Comment by ianw1974** [comment]
What OS are you attempting to build for?  Ubuntu 18.04 and 20.04 builds work.  Someone had this with Ubuntu 16.04 which I no longer support.  See: https://github.com/ianw1974/zimbra-build-scripts/issues/15 if it is related to Ubuntu 16.04.

It works for currently for Alma, CentOS, Oracle, Rocky, Ubuntu 18.04 + 20.04 as this is how I build for these on my website.

**Comment by EugenMayer** [comment]
Running on 20.04

**Comment by ianw1974** [comment]
Just running it now to check/verify, since this was working beginning of November when I built 10.0.5.

**Comment by ianw1974** [comment]
Built using podman without problems, and didn't even need to add this fix:

```
zcs-10.0.5_GA_0423.UBUNTU20_64.20231117085611.tgz
```

Are you sure you haven't edited my scripts or something that have caused this to fail, because it works perfectly fine with the current scripts in my repo and the Dockerfile.

Testing Docker so will update shortly if it's a docker thing.

**Comment by ianw1974** [comment]
Just built on my docker host as well:

```
zcs-10.0.5_GA_0423.UBUNTU20_64.20231117091323.tgz
```

I cannot replicate this problem, which means the script and Dockerfile I provide do work as intended.  If they have been changed/modified in anyway since downloading, then this could be the reason why yours isn't working.

I can add the fix, rather to my build script, so:

```
mkdir -p ~/.ivy2/cache
```

doing this way means it will create in the home directory of the user that is building Zimbra - be it root, or a normal user account.  Hard-coding to ```/root/.ivy2/cache``` would be wrong, since not everyone would build as root.  Obviously the Docker container does build as root which isn't so important since the container is destroyed afterwards and isn't going to affect a working system.  On a normal VPS or server that will remain running, then building as a normal user is preferred.

I would much prefer to find out what is causing the problem, rather than just introduce a workaround.  This shouldn't happen at all with either Ubuntu 18.04 or 20.04.  It does occur with Ubuntu 16.04 but then that is no longer supported so I didn't address that issue especially for that reason.

Even Ubuntu 18.04 has gone EOL now, although the script still supports it for the time being.  But that may change in the future.

If you can provide more info as to why this problem occurs for you - was something changed in the original Dockerfile, or what at your end is different that could cause this problem?  I for one, cannot replicate it anyway as I've just made two builds for 10.0.5 without any problems at all.

**Comment by EugenMayer** [comment]
I keep it vanilla

```yaml
name: docker

on:
  push:

env:
  IMAGE_FQDN: ghcr.io/kontextwork/zimbra-builder
  # actually that's the distro to build ON and FOR
  TARGET_DISTRO: ubuntu:20.04

jobs:
  docker:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout us
        uses: actions/checkout@v4
        with:
          path: builder
      - name: Checkout zimbra-build-scripts
        uses: actions/checkout@v4
        with:
          repository: ianw1974/zimbra-build-scripts
          ref: main
          path: zimbra-build-scripts
      - name: Set up QEMU
        uses: docker/setup-qemu-action@v2
      - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v2
      - name: Login to GHCR
        uses: docker/login-action@v2
        with:
          registry: ghcr.io
          username: ${{ github.repository_owner }}
          password: ${{ github.token }}
      - name: Build builder
        uses: docker/build-push-action@v3
        with:
          context: zimbra-build-scripts
          platforms: linux/amd64
          push: true
          build-args: |
            # actually that's the distro to build ON and FOR
            RELEASE=${{ env.TARGET_DISTRO }}
          tags: |
            ${{ env.IMAGE_FQDN }}:latest
```

and then

```yaml
    - run: |
          set -e
          cd /home/git/zimbra-build-scripts/
          ./zimbra-build-helper.sh --build-zimbra
```

depening on #60 i for sure simply can share the entire project

**Comment by ianw1974** [comment]
I expect this problem is due to using CI/CD.  You run ```ubuntu-latest```, shouldn't this be specifying ```ubuntu:20.04```?

You aren't using the Dockerfile at this point, since you are running the script inside a docker container running ubuntu-latest.  I think you should be using:

```
jobs:
  docker:
    runs-on: ubuntu:20.04
```

**Comment by ianw1974** [comment]
Alternatively you can do:

```
    - run: |
          set -e
          mkdir -p ~/.ivy2/cache
          cd /home/git/zimbra-build-scripts/
          ./zimbra-build-helper.sh --build-zimbra
```

where I added the mkdir command here for the directory you are missing.  But I expect it's due to using ubuntu-latest.

**Comment by EugenMayer** [comment]
Actually, you can now look it up yourself:

https://github.com/KontextWork/zimbra-builder/blob/main/.github/workflows/build-release.yml#L12

so i run the steps within the containrer.

And yes i do hotfix it with https://github.com/KontextWork/zimbra-builder/blob/main/.github/workflows/build-release.yml#L15 anyway and do not rely on you merging this.

Building the builder without any modifications either https://github.com/KontextWork/zimbra-builder/blob/main/.github/workflows/build-docker.yml#L35

So well, it is not big issue for me, it works nevertheless. So if you cannot reproduce it, fair enough for me, close the issue and the PR, no hard feelings

**Comment by ianw1974** [comment]
The Dockerfile was intended for using docker/podman from the command line.  The issue seems to be with the way CI/CD is being ran here, and so something doesn't match.

If the host is running Ubuntu 20.04, then the script works fine (eg: building on a VM/VPS/physical server).  If we use Docker/Podman as explained in ```README.md``` in my repository, then the container runs Ubuntu 20.04 and builds fine.  When using the script as it was intended, be it running the script, or using the Dockerfile as intended with docker/podman then all is fine.

This is why I don't want to introduce something to the scripts that is based on the way CI/CD is being ran.  I believe if CI/CD was actually loading an Ubuntu 20.04 image, and then running the script, it would work fine.  It could also be whatever the source is of that Ubuntu image, is not the one as intended in my Dockerfile - so it could be a variation of Ubuntu 20.04 or something else hence the problems.

To run ```./zimbra-build-helper.sh```, you need to be running Ubuntu 20.04 and not ```ubuntu-latest``` as in your CI/CD.  Otherwise it is running on a platform not supported by Zimbra.

**Comment by EugenMayer** [comment]
The CI runs the build inside the image that i build based on your docker file. There is no file and this is the normalized way to control the build env. I do not build on the ubuntu-latest baremetal.

**Comment by ianw1974** [comment]
That's strange, since the actual build process when all the java stuff runs, creates that directory anyway.  This is why I don't understand why it works for docker/podman with zero additions, and yet with the container image you built and use has problems.  Certainly sounds environment related.

I would have to debug it.  I generally use CI/CD on my own personal Gitlab instance with a Gitlab runner, so I can check/test it here easily enough to try and find out what the problem is.  Either that or debug on GitHub with their CI/CD.  But as mentioned, the build process should create those directories anyway, so I expect that something doesn't match right, potentially java-related or the packages related to ivy in a version mismatch for some reason.

**Comment by EugenMayer** [comment]
i do not think it is worth the effort, this issue is not that impactful, IMHO

**Comment by ianw1974** [comment]
Yeah probably right, I just have a thing about finding out why to ensure it works as intended.  I could just add it to the script and just create it anyway irrespective or whether it's needed or not for everyone else.  But I kind of feel that's more a hack fix, than finding the correct solution even if it does take me more effort and time.  But that's just how I am :)

**Comment by EugenMayer** [comment]
Do not merge my PR, since the reason behind it is unclear.

I can explain what it potentially is (it is related to the GH runner itself), but i'am neither interested in picking it apart for the GH runner (i only use gh actions for OS low level things and tend to not care too much about it).

So IMHO, close the issue, close the PR and if it does not come up again or bother us in a different way, i would duck it due to 'not worth the effort'.

**Comment by ianw1974** [comment]
Fixed here: b4821749dce48d68df22f37fbd16b8917ecead83
